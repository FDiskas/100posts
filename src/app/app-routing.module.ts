import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { WelcomeComponent } from './pages/welcome/welcome.component';

export const routes: Routes = [
  {
    path: 'welcome',
    component: WelcomeComponent,
    data: {
      name: 'Home',
    },
    pathMatch: 'full',
  },
  {
    path: '',
    redirectTo: '/welcome',
    pathMatch: 'full',
  },
  {
    path: 'posts',
    loadChildren: async () => (await import('./pages/posts/posts.module')).PostsModule,
    data: {
      name: 'Posts',
    },
    pathMatch: 'full',
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
