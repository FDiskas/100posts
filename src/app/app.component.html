<!-- Toolbar -->
<div class="container">
  <header>
    <!-- Header -->
    <app-loading></app-loading>
  </header>

  <nav>
    <!-- Nav Bar -->
  </nav>

  <main>
    <!-- Main Content -->
    <app-side-bar [title]="title" [localesList]="localesList"></app-side-bar>
  </main>

  <aside>
    <!-- Sidebar -->
  </aside>

  <footer>
    <!-- Footer -->
  </footer>
</div>
