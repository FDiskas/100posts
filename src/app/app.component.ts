import { LocationStrategy } from '@angular/common';
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  title = '100posts';

  loading = false;

  constructor(private readonly locationStrategy: LocationStrategy) {}

  localesList = [
    { location: `${window.document.location.origin}${this.locationStrategy.getBaseHref()}../en-US/`, label: 'English' },
    { location: `${window.document.location.origin}${this.locationStrategy.getBaseHref()}../lt`, label: 'Lietuvi≈≥' },
  ];
}
