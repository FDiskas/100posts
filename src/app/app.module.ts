import { BrowserModule } from '@angular/platform-browser';
import { APP_INITIALIZER, NgModule } from '@angular/core';

import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { ServiceWorkerModule } from '@angular/service-worker';
import { LayoutModule } from '@angular/cdk/layout';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatButtonModule } from '@angular/material/button';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatIconModule } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { MatGridListModule } from '@angular/material/grid-list';
import { MatCardModule } from '@angular/material/card';
import { MatMenuModule } from '@angular/material/menu';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { HttpClient, HttpClientModule, HTTP_INTERCEPTORS, HttpClientJsonpModule } from '@angular/common/http';
import { SidebarComponent } from './components/sidebar/sidebar.component';
import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { AppRoutingModule } from './app-routing.module';
import { WelcomeComponent } from './pages/welcome/welcome.component';
import { ErrorInterceptor } from './interceptors/error.interceptor';
import { loadConfig } from './interceptors/config.interceptor';
import { ConfigService } from './services/config/config.service';
import { LoadingService } from './services/loading/loading.service';
import { LoaderInterceptor } from './interceptors/loading.interceptor';
import { LoadingComponent } from './components/loading/loading/loading.component';

@NgModule({
  declarations: [AppComponent, SidebarComponent, WelcomeComponent, LoadingComponent],
  imports: [
    BrowserModule,
    AppRoutingModule,
    BrowserAnimationsModule,
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production }),
    LayoutModule,
    MatToolbarModule,
    MatButtonModule,
    MatSidenavModule,
    MatIconModule,
    MatListModule,
    MatGridListModule,
    MatCardModule,
    MatMenuModule,
    MatProgressBarModule,
    HttpClientModule,
    HttpClientJsonpModule,
  ],
  providers: [
    LoadingService,
    {
      provide: HTTP_INTERCEPTORS,
      useClass: LoaderInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: ErrorInterceptor,
      multi: true,
    },
    {
      provide: APP_INITIALIZER,
      useFactory: loadConfig,
      deps: [HttpClient, ConfigService],
      multi: true,
    },
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
