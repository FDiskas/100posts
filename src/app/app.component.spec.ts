import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'zivver'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('zivver');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.querySelector('.content span').textContent).toContain('zivver app is running!');
  });
});
