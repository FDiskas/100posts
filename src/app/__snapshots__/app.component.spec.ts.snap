// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render menu 1`] = `
<div
  id="root2"
  ng-version="10.1.3"
>
  <div
    class="container"
  >
    <header>
      <app-loading>
        
      </app-loading>
    </header>
    <nav />
    <main>
      <app-side-bar
        ng-reflect-locales-list="[object Object],[object Object"
        ng-reflect-title="100posts"
      >
        <mat-sidenav-container
          class="sidenav-container mat-drawer-container mat-sidenav-container"
        >
          
          <div
            class="mat-drawer-backdrop ng-star-inserted"
          />
          <div
            aria-hidden="true"
            class="cdk-visually-hidden cdk-focus-trap-anchor"
          />
          <mat-sidenav
            class="sidenav mat-drawer mat-sidenav ng-tns-c18-0 ng-trigger ng-trigger-transform mat-drawer-over mat-sidenav-fixed ng-star-inserted"
            fixedinviewport=""
            mode="over"
            ng-reflect-fixed-in-viewport=""
            ng-reflect-mode="over"
            ng-reflect-opened="false"
            style="top:0px;bottom:0px;box-shadow:none;visibility:hidden;0:top;1:bottom;2:box-shadow;3:visibility;"
            tabindex="-1"
          >
            <div
              class="mat-drawer-inner-container"
            >
              <mat-toolbar
                class="mat-toolbar mat-toolbar-single-row"
              >
                Navigation
              </mat-toolbar>
              
              <mat-nav-list
                class="mat-nav-list mat-list-base ng-star-inserted"
                role="navigation"
                style=""
              >
                <a
                  class="mat-list-item mat-focus-indicator"
                  color="primary"
                  href="/welcome"
                  mat-list-item=""
                  ng-reflect-router-link="welcome"
                  ng-reflect-router-link-active="active"
                  routerlinkactive="active"
                >
                  <div
                    class="mat-list-item-content"
                  >
                    <div
                      class="mat-list-item-ripple mat-ripple"
                      mat-ripple=""
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="http://localhost/welcome"
                    />
                    <div
                      class="mat-list-text"
                    />
                    Home
                  </div>
                </a>
              </mat-nav-list>
              <mat-nav-list
                class="mat-nav-list mat-list-base ng-star-inserted"
                role="navigation"
                style=""
              >
                <a
                  class="mat-list-item mat-focus-indicator"
                  color="primary"
                  href="/posts"
                  mat-list-item=""
                  ng-reflect-router-link="posts"
                  ng-reflect-router-link-active="active"
                  routerlinkactive="active"
                >
                  <div
                    class="mat-list-item-content"
                  >
                    <div
                      class="mat-list-item-ripple mat-ripple"
                      mat-ripple=""
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="http://localhost/posts"
                    />
                    <div
                      class="mat-list-text"
                    />
                    Posts
                  </div>
                </a>
              </mat-nav-list>
            </div>
          </mat-sidenav>
          <div
            aria-hidden="true"
            class="cdk-visually-hidden cdk-focus-trap-anchor"
          />
          <mat-sidenav-content
            class="mat-drawer-content mat-sidenav-content"
            fixedinviewport=""
          >
            <mat-toolbar
              class="mat-toolbar mat-primary mat-toolbar-single-row"
              color="primary"
              ng-reflect-color="primary"
            >
              <button
                aria-label="Toggle sidenav"
                class="mat-focus-indicator mat-icon-button mat-button-base"
                mat-icon-button=""
                type="button"
              >
                <span
                  class="mat-button-wrapper"
                >
                  <mat-icon
                    aria-hidden="true"
                    aria-label="Side nav toggle icon"
                    class="mat-icon notranslate material-icons mat-icon-no-color"
                    data-mat-icon-type="font"
                    role="img"
                  >
                    menu
                  </mat-icon>
                </span>
                <span
                  class="mat-button-ripple mat-ripple mat-button-ripple-round"
                  matripple=""
                  ng-reflect-centered="true"
                  ng-reflect-disabled="false"
                  ng-reflect-trigger="[object HTMLButtonElement]"
                />
                <span
                  class="mat-button-focus-overlay"
                />
              </button>
              <span>
                100posts
              </span>
              <span
                class="spacer"
              />
              <button
                aria-haspopup="true"
                class="mat-focus-indicator mat-menu-trigger mat-button mat-button-base"
                mat-button=""
                ng-reflect-menu="[object Object]"
              >
                <span
                  class="mat-button-wrapper"
                >
                  <mat-icon
                    aria-hidden="true"
                    class="mat-icon notranslate material-icons mat-icon-no-color"
                    data-mat-icon-type="font"
                    role="img"
                  >
                    language
                  </mat-icon>
                </span>
                <span
                  class="mat-button-ripple mat-ripple"
                  matripple=""
                  ng-reflect-centered="false"
                  ng-reflect-disabled="false"
                  ng-reflect-trigger="[object HTMLButtonElement]"
                />
                <span
                  class="mat-button-focus-overlay"
                />
              </button>
              <mat-menu
                class="ng-tns-c23-1 ng-star-inserted"
              >
                
              </mat-menu>
            </mat-toolbar>
            <router-outlet />
          </mat-sidenav-content>
          
        </mat-sidenav-container>
      </app-side-bar>
    </main>
    <aside />
    <footer />
  </div>
</div>
`;
